-G results in crt1.o: relocation R_X86_64_32S against `_DYNAMIC', use
-shared instead

-fPIC iso -Kpic to make it work

--- Makefile.config.in
+++ Makefile.config.in
@@ -180,7 +180,7 @@
   shliblefn = $(1:%=%.$(SHLIB_SUFFIX))
   # Solaris compiler (Sun C 5.5) can't take multiple ld options as
   # -Wl,-a,-b .  Ld sees -a,-b in that case.
-  LDFLAGS_SHLIB = -Wl,-Bdynamic -Wl,-G -Wl,-h -Wl,$(SONAME)
-  CFLAGS_SHLIB = -Kpic
+  LDFLAGS_SHLIB = -Wl,-Bdynamic -shared -Wl,-h -Wl,$(SONAME)
+  CFLAGS_SHLIB = -fPIC
 endif
 
 ifeq ($(HOST_OS),aix)
